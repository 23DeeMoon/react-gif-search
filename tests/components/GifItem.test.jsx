
import { render, screen } from "@testing-library/react"
import GifItem from "../../src/components/GifItem"

describe('Test GifItem', () => { 

    const title = 'Title'
    const url = 'https://google.com/title.jpg'
    test('should match the snapshot', () => { 
       const {container} = render(<GifItem url={url} title={title} />)
        expect(container).toMatchSnapshot()
     })

     test('should show the image and the url with the alt', () => { 
        const {container} = render(<GifItem url={url} title={title} />)
         const {src, alt} = screen.getByRole('img')
         expect(src).toBe(url)
         expect(alt).toBe(title)

      })

      test('should show the title of the img', () => { 
         render(<GifItem url={url} title={title} />)
         expect(screen.getByText(title)).toBeTruthy()
       

      })
 })