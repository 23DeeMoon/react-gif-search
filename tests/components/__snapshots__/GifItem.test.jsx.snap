// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title"
      src="https://google.com/title.jpg"
    />
    <p>
      Title
    </p>
  </div>
</div>
`;
